src/components/inputs/Select/index.js

```diff
@@ -1,123 +1,208 @@
-import React, { useState } from 'react';
+import React, { useState, useEffect } from 'react';
 import PropTypes from 'prop-types';
-import Select from 'react-select';
-import { withStyles } from '@material-ui/core';
-import { FormControl } from '../FormControl';
-import { InputLabel } from '../InputLabel';
+import TextField from '@material-ui/core/TextField';
+import Box from '@material-ui/core/Box';
+import Chip from '@material-ui/core/Chip';
+import Autocomplete from '@material-ui/lab/Autocomplete';
+import { makeStyles, CircularProgress } from '@material-ui/core';
 import { useSafeIntl } from '../../../utils/useSafeIntl';
 import { MESSAGES } from './messages';
-import { styles } from './styles';
 
-const SelectComponent = ({
+const useStyles = makeStyles(() => ({
+    chipLabel: {
+        marginTop: -2,
+    },
+    startAdornment: {
+        marginTop: -5,
+    },
+}));
+
+const SelectCustom = ({
     value,
     keyValue,
     label,
     errors,
     onChange,
     options,
+    touched,
     onBlur,
-    onFocus,
-    withMarginTop,
     multi,
     disabled,
     clearable,
-    isFocused,
-    searchable,
     required,
-    classes,
-    // noResultsText,
+    noOptionsText,
+    getOptionLabel,
+    getOptionSelected,
+    loading,
+    renderOption,
 }) => {
-    const [selectInputValue, setSelectInputValue] = useState('');
-    const hasErrors = errors.length > 0;
-    const classNames = hasErrors
-        ? [classes.select, classes.selectError]
-        : [classes.select];
+    const [selectedValue, setSelectedValue] = useState(null);
+    const [multiSelectedValue, setMultiSelectedValue] = useState([]);
     const intl = useSafeIntl();
-    const [focus, setFocus] = useState(isFocused);
+    const classes = useStyles();
+
+    const getOption = optionValue =>
+        options.find(o => `${o.value}` === `${optionValue}`);
+
+    useEffect(() => {
+        if (value) {
+            if (multi) {
+                const newSelectedValue = [];
+                const valuesList = Array.isArray(value)
+                    ? value
+                    : value.split(',');
 
+                valuesList.forEach(v => {
+                    const option = getOption(v);
+                    if (option) newSelectedValue.push(getOption(v));
+                });
+                setMultiSelectedValue(newSelectedValue);
+            } else {
+                const newSelectedValue = getOption(value);
+                if (newSelectedValue) setSelectedValue(newSelectedValue);
+            }
+        } else {
+            multi ? setMultiSelectedValue([]) : setSelectedValue(null);
+        }
+    }, [value, options]);
+
+    const handleChange = (e, newValue) => {
+        if ((!multi && !newValue) || (multi && newValue.length === 0)) {
+            return onChange(null);
+        }
+        if (multi) {
+            return onChange(newValue.map(v => v && v.value).join(','));
+        }
+        return onChange(newValue.value);
```

Comment:
Theoretically, I thiink this should be wrapped in a `useCallback` with `multi` in the dependencies array, although I agree it's unlikely `multi` will change